sopel/config/types.py

```diff
@@ -696,6 +696,9 @@ def _parse(self, value, settings, section):
         if not value:
             return self.parse(value)
 
+        if value.startswith('"') or value.endswith('"'):
+            value = value.strip('"')
```

Comment:
I would understand using this `if` for the `and` case, but why for `or`?

Reply:
You're right, there is no need for the `or`. As for the `and`... now I wonder if it is what I wanted to write, or if it is unnecessary.

Do we want to consider `"some_filename.txt` a file *with* a `"` or not? I guess not, probably? That would be a breaking change, but one we can accept for Sopel 8. People shouldn't have quotes in their filename anyway.

Reply:
I, uh… Hmm.

Common sense says "of course that should have the `"` removed!" But then I remember that if the `"` appears anywhere else in the basename, it won't be removed, and that annoys me. So maybe the suggestion below is what it should be (added handling of `'` too while I was at it):

```suggestion
        if value.startswith('"') and value.endswith('"'):
            value = value.strip('"')

        if value.startswith("'") and value.endswith("'"):
            value = value.strip("'")
```

Who are we to tell users they _can't_ set a `FilenameAttribute` to `"Who's On First?".txt` if they _really_ want it? It's perfectly legal (on common Linux distros, at least).

Reply:
Can we please just switch to TOML where all strings are quoted and escaping stupid filenames is the user's problem?
I agree with `and` until then.

Reply:
> Can we please just switch to TOML

See: all the IRC discussions about that, lol